ephemeraldata.adoc

```diff
@@ -58,6 +58,35 @@ zmprov mcf zimbraEphemeralBackendURL ssdb:<ip address|hostname>:port # substitut
 /opt/zimbra/bin/zmmigrateattrs zimbraAuthTokens zimbraCsrfTokenData zimbraLastLogonTimestamp -n 4
 ----
+
+== Additional Notes on Ephemeral Data
+
+=== Advanced Migration Options
+
+The `zmmigrateattrs` tool provides several migration options, to be used with careful consideration:
+- By default, ephemeral data is deleted from LDAP as soon as it is migrated to `SSDB`. The `-k` or `--keep-old` option will cause this data to remain in LDAP. This makes it possible to revert the ephemeral backend back to LDAP immediately after migration with no data loss. However, this data will otherwise remain in LDAP and quickly become obsolete.
+
+- The `-r` or `--dry-run` option outputs the changes to be made for each account to the console, without actually performing the migration.
+- The `-n` or `--num-threads` option specifies how many threads will be used for migration. Omitting this will result in migration happening synchronously.
+- The `-a` or `--account` option allows for migration of a comma-separated list of specific accounts. This should be used only for testing or debugging.
+- The `-d` or `--debug` option enables debug logging.
+
+If no attribute names are explicitly passed in as arguments, migration will occur for all known ephemeral attributes.
+
+=== Migration Limitations
+
+Ephemeral data migration is a one-way process. The `zmmigrateattrs` script does not support migrating data from `SSDB` back into LDAP, nor does it support migrating data between different instances of `SSDB`. This means that if the value of `zimbraEphemeralBackendURL` is reverted back to LDAP after migration, prior authentication data will become inaccessible, and all user sessions will be invalidated. If migration to a new `SSDB` backend becomes necessary, the data should be replicated to the new location prior to changing the value of `zimbraEphemeralBackendURL`.
+
+=== Changes to zmprov
+
+Due to changes in the way multi-valued ephemeral data is stored, the attributes `zimbraAuthTokens` and `zimbraCsrfTokenData` are no longer returned as part of the `zmprov ga <account>` response. The value of `zimbraLastLogonTimestamp` is returned as before.
+
+=== Migration In-Progress Flag
+
+The `zmmigrateattrs` script sets a flag in `SSDB` at the beginning of the migration process and unsets it when migration completes. When this flag is set, the server will look for an account's ephemeral data in both the `SSDB` and LDAP backends, as the data may reside in either one. In the event of an error during migration, the flag will remain set until migration is re-ran successfully. For testing or debugging purposes, this flag can be set or unset via command line using the `--set-flag` and `--unset-flag` options of `zmmigrateattrs`.
+
+=== Migration CSV Output
+
+Each run of `zmmigrateattrs` generates a CSV file in `/opt/zimbra/data/tmp/` directory. The file contains migration info for every migrated account. Migration errors are logged in a separate CSV file in the same directory.
```

Comment:
> Migration errors are logged in a separate CSV file in the same directory.

Replace with:

If any migrations fail, a cutdown CSV file report detailing only the errors is also created in the same directory.
The name(s) of the file(s) are logged at the end of the run.